### DISCUSSION TITLE
debugging information in separate files

### DISCUSSION THREAD

**Discussion by marpie** [Vector35/dwarf_import#13] [Q&A] [answered]
Is there a way to use the ubuntu-style debugging information (located in [separate files](https://sourceware.org/gdb/onlinedocs/gdb/Separate-Debug-Files.html)) using this plugin or BinaryNinja itself?

I have many executables that ship with separate -dbg*.deb that I would like to load and did not find a way in BinaryNinja itself.

**Comment by ElykDeer** [comment, marked as answer]
Theoretically, yes.

In fact, the upstream project this code comes from does exactly that.  However, it probably won't work in this form without modification.  If you'd like to give it a go yourself, check [these lines](https://github.com/Vector35/dwarf_import/blob/7fd4b4b08b5dc455094d99223d491fad608c2364/__init__.py#L30-L31) out and try providing a separate file and BV.

I'd also suggest taking a look at the branch `test_merge_upstream` which has the latest upstream code (which greatly improves import).  It's not merged yet since I made that the day before the BN-3.0 Python-refactor landed, and have yet to go through and patch things up.  When I do that, however, I'll also make sure there's a method to do what you're asking, since I need to as well! :p
